packages/headless/src/features/result-preview/result-preview-request-builder.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  HtmlRequest,
+  HtmlRequestOptions,
+} from '../../api/search/html/html-request';
+import {
+  ConfigurationSection,
+  QuerySection,
+  ResultPreviewSection,
+} from '../../state/state-sections';
+
+export type StateNeededByHtmlEndpoint = ConfigurationSection &
+  ResultPreviewSection &
+  Partial<QuerySection>;
+
+export function buildResultPreviewRequest(
+  state: StateNeededByHtmlEndpoint,
+  options: HtmlRequestOptions
+): HtmlRequest {
+  const {search, accessToken, organizationId} = state.configuration;
```

Comment:
Unless I am mistaken, the API should support * all * the query parameters for this endpoint. Any reason why it's not the case here ? 

I think we'd wanna refactor/share much of the code from `buildSearchRequest` actually.

Reply:
I'm not so sure. I asked Alexis about this, and did some tests. From what I see, only the `q` is used, to highlight keywords.

Example: https://static.cloud.coveo.com/searchui/v2.10081/index.html#q=universe%20program&t=All&sort=relevancy&f:@author=[Gregory%20Laporte]

Thread with Alexis: https://coveo.slack.com/archives/G016TA2G485/p1615471295013600?thread_ts=1615417082.013500&cid=G016TA2G485